DesktopModules/Vanjaro/UXManager/Extensions/Menu/Pixabay/Controllers/SettingController.cs

```diff
@@ -17,33 +18,51 @@ public class SettingController : UIEngineController
-            Settings.Add("HasApiKey", new UIData { Name = "HasApiKey", Options = HasApiKey });
+            
+            string Site_ApiKey = PortalController.GetEncryptedString("Vanjaro.Integration.Pixabay.ApiKey", portalId, Config.GetDecryptionkey());
+            string Host_ApiKey = HostController.Instance.GetEncryptedString("Vanjaro.Integration.Pixabay.ApiKey", Config.GetDecryptionkey());
+            
+            Settings.Add("IsSuperUser", new UIData { Name = "IsSuperUser", Options = UserController.Instance.GetCurrentUserInfo().IsSuperUser });
+            Settings.Add("ApplyTo", new UIData { Name = "ApplyTo", Options = false });
+            Settings.Add("Host_ApiKey", new UIData { Name = "Host_ApiKey", Value = Host_ApiKey });
+            Settings.Add("Host_HasApiKey", new UIData { Name = "Host_HasApiKey", Options = string.IsNullOrEmpty(Host_ApiKey) ? false : true });
+            Settings.Add("Site_ApiKey", new UIData { Name = "Site_ApiKey", Value = Site_ApiKey });
+            Settings.Add("Site_HasApiKey", new UIData { Name = "Site_HasApiKey", Options = string.IsNullOrEmpty(Site_ApiKey) ? false : true });
             return Settings.Values.ToList();
         }
 
         [HttpPost]
         public bool Save(dynamic Data)
         {
-            if (Vanjaro.Core.Providers.Pixabay.IsValid(Data.ToString()))
-            {
-                PortalController.UpdateEncryptedString(PortalSettings.PortalId, "Vanjaro.Integration.Pixabay", Data.ToString(), Config.GetDecryptionkey());
+                if (bool.Parse(Data.ApplyTo.ToString()))
+                {
+                    if (Vanjaro.Core.Providers.Pixabay.IsValid(Data.Host_ApiKey.ToString()))
+                    {
+                        HostController.Instance.UpdateEncryptedString("Vanjaro.Integration.Pixabay.ApiKey", Data.Host_ApiKey.ToString(), Config.GetDecryptionkey());
+                    }
+                    else
+                        return false;
+                }
+                else 
+                {
+                    if (Vanjaro.Core.Providers.Pixabay.IsValid(Data.Site_ApiKey.ToString()))
+                    {
+                        PortalController.UpdateEncryptedString(PortalSettings.PortalId, "Vanjaro.Integration.Pixabay.ApiKey", Data.Site_ApiKey.ToString(), Config.GetDecryptionkey());
+                    }
+                    else
+                        return false;
+                } 
                 return true;
-            }
-            return false;
+            
         }
 
-        [HttpGet]
-        public string Delete()
+        [HttpPost]
+        public string Delete(dynamic Data)
         {
-            PortalController.UpdateEncryptedString(PortalSettings.PortalId, "Vanjaro.Integration.Pixabay", string.Empty, Config.GetDecryptionkey());
+            if (bool.Parse(Data.ToString()))
+                HostController.Instance.UpdateEncryptedString("Vanjaro.Integration.Pixabay.ApiKey", string.Empty, Config.GetDecryptionkey());
```

Comment:
why are you change the key name **Vanjaro.Integration.Pixabay** to this **Vanjaro.Integration.Pixabay.ApiKey**

Reply:
I believe this (Vanjaro.Integration.Pixabay.ApiKey) make more sense that what key we are going to save

Reply:
Pixabay Key has been changed